x/programs/examples/simulator/cmd/program.go

```diff
@@ -0,0 +1,177 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/ava-labs/hypersdk/crypto/ed25519"
+	"github.com/ava-labs/hypersdk/utils"
+	"github.com/ava-labs/hypersdk/x/programs/runtime"
+	"github.com/spf13/cobra"
+)
+
+const (
+	HRP       = "sim_key_"
+	keyPrefix = 0x1
+)
+
+// example cost map
+var costMap = map[string]uint64{
+	"ConstI32 0x0": 1,
+	"ConstI64 0x0": 2,
+}
+
+var programCmd = &cobra.Command{
+	Use: "program",
+	RunE: func(*cobra.Command, []string) error {
+		return ErrMissingSubcommand
+	},
+}
+
+var programCreateCmd = &cobra.Command{
+	Use:   "create [path to wasm program]",
+	Short: "Creates a program from a wasm file, calls init_program and returns the program ID",
+	PreRunE: func(cmd *cobra.Command, args []string) (err error) {
+		if len(args) != 1 {
+			return ErrInvalidArgs
+		}
+		if callerAddress == "" {
+			return ErrMissingAddress
+		}
+		return nil
+	},
+	RunE: func(_ *cobra.Command, args []string) error {
+		filePath := args[0]
+		fileBytes, err := os.ReadFile(filePath)
+		if err != nil {
+			return err
+		}
+
+		pk, err := getPublicKey(db, callerAddress)
+		if err != nil {
+			return err
+		}
+
+		// getKey(db, pk)
+		programId, err := initalizeProgram(fileBytes, pk)
+		if err != nil {
+			return err
+		}
+
+		err = runtime.SetProgram(db, programId, pk, fileBytes)
+		if err != nil {
+			return err
+		}
+		utils.Outf("{{green}}create program action successful program id:{{/}} %v\n", programId)
+		return nil
+	},
+}
+
+func initalizeProgram(programBytes []byte, caller ed25519.PublicKey) (uint64, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	runtime := runtime.New(log, runtime.NewMeter(log, maxFee, costMap), db, caller)
+	defer runtime.Stop(ctx)
+
+	programID, err := runtime.Create(ctx, programBytes)
```

Comment:
the ID here should be a successful transaction as the result of something like create_program action. Flow.

- create_program action -> programID.
- all invoke calls to the program reference that id. The ID will not be created by wasm in the VM so we should not set that pattern in the examples IMO.